src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -648,8 +734,8 @@ public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node
             Debug.Assert(IsConditionalState);
             if (slot > 0)
             {
-                this.StateWhenTrue[slot] = false;
-                this.StateWhenFalse[slot] = true;
+                Assign(ref this.StateWhenTrue, slot, false);
+                Assign(ref this.StateWhenFalse, slot, true);
```

Comment:
>Assign(ref this.StateWhenFalse, slot, true); [](start = 16, length = 44)

This feels wrong, this shouldn't change state of members because no assignment happens during 'is' pattern evaluation.